subprojects/base-services/src/main/java/org/gradle/internal/util/ClassUtils.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.util;
+
+import org.gradle.internal.UncheckedException;
+
+import java.lang.reflect.Constructor;
+
+public class ClassUtils {
+    /**
+     * This is intended to be a equivalent of deprecated {@link Class#newInstance()}.
+     */
+    public static <T> T newInstance(Class<T> c) {
```

Comment:
I think it would make sense to add that method to `JavaReflectionUtils`, given that the class already contains a method `newInstanceOrFallback`. WDYT?

Reply:
Thanks, I failed to find a class to put this method into so I created a new class. Already moved to `JavaReflectionUtils`.